mm2src/coins/lightning/ln_events.rs

```diff
@@ -192,21 +250,40 @@ impl LightningEventHandler {
             },
             false => HTLCStatus::Failed,
         };
-        let mut payments = self.inbound_payments.lock();
-        match payments.entry(payment_hash) {
-            Entry::Occupied(mut e) => {
-                let payment = e.get_mut();
-                payment.status = status;
-                payment.preimage = Some(payment_preimage);
-                payment.secret = payment_secret;
-            },
-            Entry::Vacant(e) => {
-                e.insert(PaymentInfo {
+        let persister = self.persister.clone();
+        match purpose {
+            PaymentPurpose::InvoicePayment { .. } => spawn(async move {
+                if let Ok(Some(mut payment_info)) = persister
+                    .get_payment_from_sql(payment_hash)
+                    .await
+                    .error_log_passthrough()
+                {
+                    payment_info.preimage = Some(payment_preimage);
+                    payment_info.status = HTLCStatus::Succeeded;
+                    payment_info.amt_msat = Some(amt);
+                    payment_info.last_updated = now_ms() / 1000;
+                    if let Err(e) = persister.add_or_update_payment_in_sql(payment_info).await {
+                        log::error!("{}", e);
+                    }
```

Comment:
I do think we have something named `log_passthrough` for things like that

Reply:
If I don't use the result I will get a clippy error that the result must be used, so instead of using `let _ = persister.add_or_update_payment_in_sql(payment_info).await.error_log_passthrough()` I just logged the error regularly. Maybe I can just keep this but write a better error message, it will be more useful this way.